### PR TITLE
(Fix) Split userAddressesCount and userLastUserName

### LINKED ISSUE #78: Addresses considered in functions that not necessarily should
- **What is it?** (leave one option)
    * `(Bug)` report (i.e. something doesn't work as it should)

- **What problem have you encountered (provide steps to reproduce if possible) / What feature are you talking about?**

The function [user_last_used_name](https://github.com/poanetwork/poa-popa/blob/master/blockchain/contracts/ProofOfPhysicalAddress.sol#L165) returns the last name submited regardless of it being confirmed.

The function [user_addresses_count](https://github.com/poanetwork/poa-popa/blob/master/blockchain/contracts/ProofOfPhysicalAddress.sol#L173) also considers unconfirmed addresses.

- **What is expected behaviour**

Only consider confirmed addresses.

### PR DESCRIPTION
**Pull request by phahulin**
- **What is it?** (leave one option)
    * Bug `(Fix)`

- **What was the root cause of the problem originally / what feature was missing?**
Functions `userAddressesCount` and `userLastUserName` counted all `physicalAddresses`, both confirmed and not confirmed, which might lead to confusion.

- **How does this pull request solve it (in broad terms)?**
Split each of these functions in two variants, e.g. `userLastSubmittedName` and `userLastConfirmedName` to clearly distinguish between the two.
Also update `contract-output.json` file used for tests

- **Does it close any open issues?**
Closes #78 

- **Quick checklist**
    - [X] `npm run lint` shows no errors
    - [X] docs were updated where necessary OR they don't need to be updated
    - [X] tests were updated where necessary OR they don't need to be updated

### CODE REVIEW

File: blockchain/contracts/ProofOfPhysicalAddress.sol [hunk 1]
```diff
@@ -154,20 +154,47 @@ contract ProofOfPhysicalAddress {
...
     }
 
-    // if user does not exist, returns 0
-    function userAddressesCount(address wallet)
+    // returns name from the last confirmed address. If no addresses were confirmed returns ''
+    function userLastConfirmedName(address wallet)
+    public constant returns (string)
+    {
+        require(userExists(wallet));
+        for (uint256 ai = users[wallet].physicalAddresses.length-1; ai >= 0; ai -= 1) {
```
Comment (fvictorio) [inline blockchain/contracts/ProofOfPhysicalAddress.sol]: Does this work? `ai` is unsigned, so it should always be greater or equal than zero.

Comment (phahulin) [reply blockchain/contracts/ProofOfPhysicalAddress.sol]: Why do you think it might become negative?

Comment (fvictorio) [reply blockchain/contracts/ProofOfPhysicalAddress.sol]: Let's say the length is 1. So `ai` is initialized to 0. And suppose the block doesn't finish in that iteration.

In the next iteration, `ai` would be substracted 1. But, as far as I know, substracting 1 from 0 in an unsigned integer will give it the max value of the integer, not -1.

What I'm saying is: if you call this method with one registered but unconfirmed address, it should get in an infinite loop and run out of gas. (I guess it's easy to try out.)

Does that make sense?

Comment (phahulin) [reply blockchain/contracts/ProofOfPhysicalAddress.sol]: Yes, you're right. Please check now

**Review by fvictorio — APPROVED** [review]

**Pull request merged**